watershed_workflow/river_mesh.py

```diff
@@ -362,6 +364,7 @@ def create_river_corridor(river, width):
     corr3 = shapely.geometry.Polygon(corr3_p)
 
     # check if the points on the river corridor are same as calculated theoretically
+    is_broken, n = _isExpectedNumPoints(corr3, river)
     if not _isExpectedNumPoints(corr3, river):
```

Comment:
This isn't right -- maybe it should be:

    is_broken, n = _isExpectedNumPoints()
    if is_broken:

But really, I don't like having a function called `_isExpectedNumPoints()` that doesn't return a single boolean.  So if you want n, pass it in, e.g.

    def _isExpectedNumPoints(corr3, river, n):
         n = # do complicated thing
         return True/False

    n = 0
    if not _isExpectedNumPoints(...., n):
        print(n)

Reply:
Does this one still need to be fixed?

Reply:
fixed it with your suggested change.